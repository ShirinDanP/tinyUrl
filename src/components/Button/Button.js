import React from 'react';
import classNames from 'classnames';
import { PropTypes } from 'prop-types';

import styles from './Button.module.css';

export const Button = ({ onClick, children, className }) =>
  React.createElement(
    'button',
    {
      onClick,
      className: classNames(styles.button, className),
    },
    children
  );

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
};
