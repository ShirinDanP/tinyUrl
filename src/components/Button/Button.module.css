.button {
    padding: 10px 8px 10px 8px;
    font-size: 13px;
    text-decoration: none;
    color: white;
    background-color: #407ab1;
    cursor: pointer;
    user-select: none;
    outline: none;
    border-radius: 2px;
    border: 2px solid #407ab1;
    margin-top: 1rem;
}

.button:hover {
    background-color: gray;
    color: black;
    border: 2px solid black;
}

.button:active {
    background-color: #275882;
    color: white;
}