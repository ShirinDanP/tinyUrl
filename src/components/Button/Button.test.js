import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import { Button } from './Button';

configure({ adapter: new Adapter() });
afterEach(cleanup);

describe('Button', () => {
  test('should render with no errors', () => {
    const { container } = render(<Button>Test text</Button>);

    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveTextContent('Test text');
    expect(container.firstChild.nodeName).toBe('BUTTON');
  });

  it('Test click event', () => {
    const mockCallBack = jest.fn();
    const button = shallow(
      <Button onClick={mockCallBack}>Onclick works!</Button>
    );
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
