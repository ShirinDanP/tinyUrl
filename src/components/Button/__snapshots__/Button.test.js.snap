// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with no errors 1`] = `
<div>
  <button
    class="button"
  >
    Test text
  </button>
</div>
`;
